File: Medium/2026-05-04-48-Rotate-Image/solution.js
var rotate = function(matrix) {
    const n = matrix.length;
    
    // First, transpose the matrix
    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]];
        }
    }
    
    // Then, reverse each row
    for (let i = 0; i < n; i++) {
        matrix[i].reverse();
    }
};

File: Medium/2026-05-04-48-Rotate-Image/solution.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        n = len(matrix)
        # First, transpose the matrix
        for i in range(n):
            for j in range(i, n):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
        
        # Then, reverse each row
        for i in range(n):
            matrix[i].reverse()